fix: TRTLLM build from source was not repsecting the wheel installation directory

nv-kmcgill53 opened the pull request:
#### Overview:

<!-- Describe your pull request here. Please read the text below the line, and make sure you follow the checklist.-->

When invoking `build.sh` for the trtllm framework with the intention of building trtllm from source and then installing that build into the dynamo container, the `--tensorrtllm-pip-wheel-dir` was not being respected. Additionally, users would get the dynamo specified nixl version when building the trtllm wheel, not the nixl version which is specified in the trtllm repository itself. This causes build issues when these versions drift far apart.

#### Details:

<!-- Describe the changes made in this PR. -->

#### Where should the reviewer start?

<!-- call out specific files that should be looked at closely -->

#### Related Issues: (use one of the action keywords Closes / Fixes / Resolves / Relates to)

- closes DIS-1036


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

* **New Features**
  * Enhanced TRTLLM build script with improved handling for download, install from wheel, and build-from-source workflows.
  * Better error messages and validation for flag combinations.

* **Tests**
  * Added comprehensive test suite covering TRTLLM installation and build scenarios including edge cases and invalid combinations.

* **Chores**
  * Removed unused installation script copying logic from build process.

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

nv-kmcgill53 committed:
working thread: Respecting the TENSORRTLLM_PIP_WHEEL_DIR during source builds

nv-kmcgill53 committed:
Removed modifying the trtllm build script to overwrite the install_nixl.sh script

nv-kmcgill53 committed:
added tests for the trtllm framework build.sh interface

dmitry-tokarev-nv reviewed (changes requested):
I didn't find these tests execution in the pre-merge checks. I think the tests need markers.

nv-kmcgill53 committed:
Apply suggestions from code review

Co-authored-by: Dmitry Tokarev <dtokarev@nvidia.com>
Signed-off-by: Kyle McGill <101670481+nv-kmcgill53@users.noreply.github.com>

nv-kmcgill53 committed:
pre-commit fixes

nv-kmcgill53 committed:
adding correct pytest markers

nv-kmcgill53 committed:
Now download the commit to a temporary directory for being able to execute build.py

Because build.sh is not included in the container, we download
the HEAD of the current branch so that we have the entire
repository to be able to call the correct build.sh with
all the dependencies it needs to execute.

nv-kmcgill53 committed:
Using DYNAMO_GIT_COMMIT environment variable as default

nv-kmcgill53 committed:
changing variable to DYNAMO_COMMIT_SHA

nv-kmcgill53 committed:
adding DYNAMO_COMMIT_SHA to the trtllm runtime container

nv-kmcgill53 committed:
adding DYNAMO_COMMIT_SHA to all the containers so that tests dont fail

nv-kmcgill53 committed:
defualting the commit sha if its defined in the environment, else look at the repository HEAD

nv-kmcgill53 committed:
protecting all git commands for testing build.sh

nv-kmcgill53 committed:
passing dummy values for testing

nv-kmcgill53 committed:
adding DYNAMO_COMMIT_SHA to the base container Dockerfile

dmitry-tokarev-nv commented on container/build.sh:
```diff
@@ -26,13 +26,13 @@ RUN_PREFIX=
 
 # if COMMIT_ID matches a TAG use that
-current_tag=$(git describe --tags --exact-match 2>/dev/null | sed 's/^v//') || true
+current_tag=${commit_tag:-$($(git describe --tags --exact-match 2>/dev/null | sed 's/^v//') || true)}
```
is commit_tag defined anywhere?

nv-kmcgill53 replied:
The issue is that I am trying to run `build.sh` without the git metadata being in the same directory. I just download the repository as a tarball rather than `git clone`. So this is a WAR to say, If it's not defined, then do the script to populate the `current_tag` , if it is, just use the value in the env.

dmitry-tokarev-nv replied:
```
➜  dynamo git:(v0.6.1) current_tag=${commit_tag:-$(git describe --tags --exact-match 2>/dev/null | sed 's/^v//' || true)}
➜  dynamo git:(v0.6.1) echo $current_tag                                                                                 
0.6.1
➜  dynamo git:(v0.6.1) git checkout main
Previous HEAD position was 38ed7de1e chore: Update missing license headers in rust attributions (#4146)
Switched to branch 'main'
Your branch is behind 'origin/main' by 28 commits, and can be fast-forwarded.
  (use "git pull" to update your local branch)
➜  dynamo git:(main) current_tag=${commit_tag:-$(git describe --tags --exact-match 2>/dev/null | sed 's/^v//' || true)}
➜  dynamo git:(main) echo $current_tag                                                                                 

➜  dynamo git:(main)

dmitry-tokarev-nv commented on container/build.sh:
```diff
@@ -26,13 +26,13 @@ RUN_PREFIX=
 
 # Get latest TAG and add COMMIT_ID for dev
-latest_tag=$(git describe --tags --abbrev=0 "$(git rev-list --tags --max-count=1)" | sed 's/^v//') || true
+latest_tag=${latest_tag:-$($(git describe --tags --abbrev=0 "$(git rev-list --tags --max-count=1)" | sed 's/^v//') || true)}
```
I think it does. I tested locally

```
dynamo git:(main) latest_tag=${latest_tag:-$(git describe --tags --abbrev=0 "$(git rev-list --tags --max-count=1)" | sed 's/^v//' || true)} 
➜  dynamo git:(main) echo $latest_tag                                                                                                            
0.6.1
```

nv-kmcgill53 committed:
Update container/build.sh

Co-authored-by: Dmitry Tokarev <dtokarev@nvidia.com>
Signed-off-by: Kyle McGill <101670481+nv-kmcgill53@users.noreply.github.com>

nv-kmcgill53 committed:
Update container/build.sh

Co-authored-by: Dmitry Tokarev <dtokarev@nvidia.com>
Signed-off-by: Kyle McGill <101670481+nv-kmcgill53@users.noreply.github.com>

nv-kmcgill53 committed:
typo
